cloudify_agent/operations.py

```diff
@@ -195,17 +195,14 @@ def _get_broker_url(ctx, agent):
 
 
 def create_new_agent_dict(old_agent):
-    new_agent = copy.deepcopy(old_agent)
-    fields_to_delete = ['name', 'queue', 'workdir', 'agent_dir', 'envdir',
-                        'manager_ip', 'package_url', 'source_url'
-                        'manager_port', 'broker_url', 'broker_port',
-                        'broker_ip', 'key', 'version']
-    for field in fields_to_delete:
-        if field in new_agent:
-            del(new_agent[field])
+    new_agent = {}
     new_agent['name'] = utils.internal.generate_new_agent_name(
         old_agent['name'])
     new_agent['remote_execution'] = True
+    fields_to_copy = ['windows', 'ip', 'basedir', 'user']
```

Comment:
Otherwise this line will fail:
https://github.com/cloudify-cosmo/cloudify-agent/blob/master/cloudify_agent/installer/config/configuration.py#L145
The more I think about it the more shaky it seems.
I think it might be a good idea to change this to use standard priorities from configuration module, just skipping runtime properties.